hyperspy/axes.py

```diff
@@ -330,12 +380,6 @@ def index2value(self, index):
         else:
             return self.axis[index]
 
-    def set_index_from_value(self, value):
-        self.index = self.value2index(value)
-        # If the value is above the limits we must correct the value
-        if self.continuous_value is False:
-            self.value = self.index2value(self.index)
```

Comment:
`update_value` should have been private as it had no external use. I'll deprecate only `set_index_from_value`.